test/e2e/util.go

```diff
@@ -174,3 +189,41 @@ func getDeployment(t *testing.T, client client.Client, name types.NamespacedName
 	}
 	return &dep, nil
 }
+
+// Send HTTPS Request to a route and check what is returned in the Response
+func getHttpResponse(t *testing.T, route *routev1.Route) (http.Header, int, string, error) {
+	t.Helper()
+	if len(route.Spec.Host) == 0 {
+		return nil, 0, "", fmt.Errorf("route.Spec.Host is empty, cannot test route")
+	}
+	request, err := http.NewRequest("GET", "https://"+route.Spec.Host, nil)
+	if err != nil {
+		return nil, 0, "", fmt.Errorf("error creating HTTPS request %v: %v", request, err)
+	}
+
+	// Send the HTTPS request
+	timeout, _ := time.ParseDuration("10s")
+	client := &http.Client{
+		Timeout: timeout,
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		},
+	}
+
+	// Get the response
+	response, err := client.Do(request)
+	if err != nil {
+		return response.Header, response.StatusCode, "", fmt.Errorf("error receiving HTTP response %v", err)
```

Comment:
If there is an error we can't deref `response`. We should return the zero value for the header and status code types.